src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -584,6 +584,23 @@ private void CompileNamedType(NamedTypeSymbol containingType)
                 }
             }
 
+            foreach (var member in members)
+            {
+                if (member is SourcePropertySymbolBase property)
+                {
+                    // PROTOTYPE(semi-auto-props): This can be optimized by checking for field keyword syntactically first.
+                    // If we have field keyword, there is no need to call MarkBackingFieldAsCalculated.
```

Comment:
>If we have field keyword, there is no need to call MarkBackingFieldAsCalculated. [](http://example.com/codeflow?start=23&length=80)

Sorry, I think this doesn't quite match what I said: "We can avoid checking accessors that failed syntactic check for a field keyword." In other words, if an accessor doesn't have a ```field``` keyword according to the syntactic check, we don't care if it passes the filter or not. Binding that accessor is not supposed to synthesize the backing field.

Reply:
Obviously my prototype comment doesn't make any sense.
I had a typo, meant to say "don't have field keyword", which is still wrong per your clarification but is what I initially understood. Fixed in the last commit. Thanks!